pkg/virt-launcher/virtwrap/nichotplug_test.go

```diff
@@ -141,6 +149,79 @@ var _ = Describe("nic hotplug on virt-launcher", func() {
 	)
 })
 
+var _ = Describe("domain network interfaces resources", func() {
+
+	DescribeTable("are ignored when",
+		func(interfaceResourceRequest string, vmiSpecIfaces []v1.Interface) {
+			vmi := &v1.VirtualMachineInstance{}
+			vmi.Spec.Domain.Devices.Interfaces = vmiSpecIfaces
+			if interfaceResourceRequest != "" {
+				vmi.Spec.Domain.Resources.Requests = k8sv1.ResourceList{
+					resource.ResourceInterface: k8sresource.MustParse(interfaceResourceRequest),
+				}
+			}
+			domainSpec := &api.DomainSpec{}
+			countCalls := 0
+			_, _ = withNetworkIfacesResources(vmi, domainSpec, func(v *v1.VirtualMachineInstance, s *api.DomainSpec) (cli.VirDomain, error) {
+				countCalls++
+				return nil, nil
+			})
+			// The counter tracks the tested function behavior.
+			// It is expected that the callback function is called only once when there is no need
+			// to add placeholders interfaces.
+			Expect(countCalls).To(Equal(1))
+		},
+		Entry("no interface resource-request is present", "", []v1.Interface{{}}),
+		Entry("the interface resource-request is zero", "0", []v1.Interface{{}}),
+		Entry("the interface resource-request is less than defined interfaces", "1", []v1.Interface{{}, {}}),
+	)
+
+	DescribeTable("are reserved when",
+		func(interfaceResourceRequest string, vmiSpecIfaces []v1.Interface) {
+			vmi := &v1.VirtualMachineInstance{}
+			vmi.Spec.Domain.Devices.Interfaces = vmiSpecIfaces
+			if interfaceResourceRequest != "" {
+				vmi.Spec.Domain.Resources.Requests = k8sv1.ResourceList{
+					resource.ResourceInterface: k8sresource.MustParse(interfaceResourceRequest),
+				}
+			}
+			domainSpec := &api.DomainSpec{}
+			for range vmiSpecIfaces {
+				domainSpec.Devices.Interfaces = append(domainSpec.Devices.Interfaces, api.Interface{})
+			}
+
+			ctrl := gomock.NewController(GinkgoT())
+			mockDomain := cli.NewMockVirDomain(ctrl)
+			domxml, err := xml.MarshalIndent(domainSpec, "", "\t")
+			Expect(err).ToNot(HaveOccurred())
+			mockDomain.EXPECT().GetXMLDesc(libvirt.DomainXMLFlags(2)).Return(string(domxml), nil)
+
+			countCalls := 0
+			_, err = withNetworkIfacesResources(vmi, domainSpec, func(v *v1.VirtualMachineInstance, s *api.DomainSpec) (cli.VirDomain, error) {
+				// Tracking the behavior of the tested function.
+				// It is expected that the callback function is called twice when placeholders are needed.
+				// The first time it is called with the placeholders in place.
+				// The second time it is called without the placeholders.
+				countCalls++
+				if countCalls == 1 {
+					ifaceRequest, err := strconv.Atoi(interfaceResourceRequest)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(s.Devices.Interfaces).To(HaveLen(ifaceRequest))
+				} else {
+					Expect(s.Devices.Interfaces).To(HaveLen(len(vmiSpecIfaces)))
```

Comment:
It would be preferable to match against the actual `Interface`. 

Out of the top of my head, it is not that harder. Could you do it ? 

Disregard this comment if it is too much trouble.

Reply:
I can use a strong `Equal` to assure the interfaces revert back to the original (in the last config creation).

To show the placeholder interface with the existing defined ones is more tricky.
The result, in my view, is too complex and adds a lot of logic I would prefer to avoid.
It would be better if that is a must, to create individual tests and define the exact expectation (without the table test).

Here is a snippet of how it looks:
```go
if countCalls == 1 {                                                     
        ifaceRequest, err := strconv.Atoi(interfaceResourceRequest)      
        Expect(err).NotTo(HaveOccurred())                                
        expectedIfaces := append([]api.Interface{}, originalDomainSpec.Devices.Interfaces...)
        for i := 0; i < ifaceRequest-len(originalDomainSpec.Devices.Interfaces); i++ {
                expectedIfaces = append(expectedIfaces, NewInterfacePlaceholder(i))
        }                                                                
        Expect(s.Devices.Interfaces).To(Equal(expectedIfaces))           
} else {                                                                 
        Expect(s.Devices.Interfaces).To(Equal(originalDomainSpec.Devices.Interfaces))
}      
```

I will go with a partial one and avoid the logic complexity.
Let me know if this is a good enough compromise.